tests/test_cli.py

```diff
@@ -330,3 +344,29 @@ def test_download_a_file(tmp_path, patched_repo_and_user, repo):
     os.chdir(working_dir)
     main(["dl", "hello.md"])
     assert (working_dir / "hello.md").read_bytes() == b"awesome content"
+
+
+def test_download_a_file_from_public_repo(
+    tmp_path, patched_another_repo_and_user, outside_repo, requests_mock, patched_repo_and_user
+):
+    requests_mock.get(
+        "https://raw.githubusercontent.com/another_awesome/repository/main/hello.md", content=b"awesome content"
+    )
+    working_dir = tmp_path / "working_dir"
+    working_dir.mkdir()
+    os.chdir(working_dir)
+    main(["dl", "https://github.com/another_awesome/repository/blob/main/hello.md"])
+    assert (working_dir / "hello.md").read_bytes() == b"awesome content"
+
+
+def test_download_a_file_from_public_repo_rasie_error(
+    tmp_path, patched_another_repo_and_user, outside_repo, requests_mock, patched_repo_and_user
+):
+    requests_mock.get("https://raw.githubusercontent.com/another_awesome/repository/main/hello.md", status_code=400)
+    working_dir = tmp_path / "working_dir"
+    working_dir.mkdir()
+    os.chdir(working_dir)
```

Comment:
not a good practice to change the dir in this way. Use `monkeypatch.chdir` so it's undo automatically on teardown (even if the test fails or errored).  https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.chdir